umh-core/pkg/fsm/redpanda_monitor/manager.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redpanda_monitor
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/config"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/constants"
+	public_fsm "github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/fsm"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/logger"
+	"github.com/united-manufacturing-hub/united-manufacturing-hub/umh-core/pkg/metrics"
+)
+
+// RedpandaMonitorManager is the FSM manager for the redpanda monitor instance
+type RedpandaMonitorManager struct {
+	*public_fsm.BaseFSMManager[config.RedpandaMonitorConfig]
+}
+
+// RedpandaMonitorManagerSnapshot extends the base manager snapshot to hold any redpanda monitor-specific info
+type RedpandaMonitorManagerSnapshot struct {
+	*public_fsm.BaseManagerSnapshot
+}
+
+// Ensure it satisfies fsm.ObservedStateSnapshot
+func (b *RedpandaMonitorManagerSnapshot) IsObservedStateSnapshot() {}
+
+// NewRedpandaMonitorManager constructs a manager.
+func NewRedpandaMonitorManager() *RedpandaMonitorManager {
+	managerName := fmt.Sprintf("%s_%s", logger.ComponentRedpandaMonitorManager, constants.RedpandaMonitorServiceName)
+
+	baseMgr := public_fsm.NewBaseFSMManager[config.RedpandaMonitorConfig](
+		managerName,
+		"/dev/null", // no actual S6 base dir needed for a pure monitor
```

Comment:
Is this correct to have /dev/null here ?

Reply:
Yes